lib/archethic/contracts/worker.ex

```diff
@@ -40,10 +43,56 @@ defmodule Archethic.Contracts.Worker do
     GenServer.cast(via_tuple(resolved_address), {:execute, tx, recipient})
   end
 
-  def init(contract = %Contract{}) do
+  @doc """
+  Return true if a worker exists for the genesis address
+  """
+  @spec exists?(genesis_address :: Crypto.prepended_hash()) :: boolean()
+  def exists?(genesis_address),
+    do: genesis_address |> via_tuple() |> GenServer.whereis() != nil
+
+  @doc """
+  Start a new worker for the genesis address
+  """
+  @spec new(genesis_address :: Crypto.prepended_hash(), contract :: Contract.t()) ::
+          DynamicSupervisor.on_start_child()
+  def new(genesis_address, contract) do
+    DynamicSupervisor.start_child(
+      ContractSupervisor,
+      {__MODULE__, contract: contract, genesis_address: genesis_address}
+    )
+  end
+
+  @doc """
+  Stop a worker for the genesis address
+  """
+  @spec stop(genesis_address :: Crypto.prepended_hash()) :: :ok | {:error, :not_found}
+  def stop(genesis_address) do
+    case genesis_address |> via_tuple() |> GenServer.whereis() do
+      nil ->
+        :ok
+
+      pid ->
+        Logger.info("Stop smart contract at #{Base.encode16(genesis_address)}")
+        DynamicSupervisor.terminate_child(ContractSupervisor, pid)
+    end
+  end
```

Comment:
I think it would be better to decouple worker to the supervisor which holds it.
Maybe we can use a WorkerSupervisor (like for Connection), which does the action to spawn and terminate worker, or let the loader doing it.